### ISSUE TITLE
PR #26088 introduces a breaking change, mentioned as a "new feature" in 2.5.0-RC1

### ISSUE THREAD

**Issue by dan-lind** [spring-projects/spring-boot#28675]
I'm in the process of upgrading our project from 2.4.2 to 2.5.6. 
While in the progress of testing this, we started seeing a lot or errors saying "This ApplicationConversionService cannot be modified". 

It looks like this is caused by us using `ApplicationConversionService.getSharedInstance().addConverter()` and a change that was introduced in https://github.com/spring-projects/spring-boot/issues/26088

The top bullet in the RC1 mentions this as a new feature. 
https://github.com/spring-projects/spring-boot/releases?page=3

Looks like a breaking change to me? I think it would be nice if it was properly documented as such, and with at least a pointer to the documentation of how to add converters instead.

**Comment by wilkinsona** [comment]
Good catch, thank you. This should have been mentioned in the [release notes](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.5-Release-Notes).

**Comment by bclozel** [comment]
Sorry we forgot about this one. At this point, updating the release notes for Spring Boot 2.5 isn't helpful anymore.
